From 3d3a13c561e858853af5c601373be3ea0746f58c Mon Sep 17 00:00:00 2001
From: Guillaume Desmottes <guillaume.desmottes@collabora.co.uk>
Date: Thu, 28 Aug 2014 14:19:30 +0200
Subject: server: exit early if we failed to create McdService

mcd_service_new() can return NULL if we didn't manage to connect to D-Bus.
Best to handle this gracefully rather than crashing horribly.

This is the top crasher on Ubuntu btw.
https://errors.ubuntu.com/problem/7e2a5f9a379f34a347895782ae948177926a1932

diff --git a/server/mc-server.c b/server/mc-server.c
index 6e77950..aa6c696 100644
--- a/server/mc-server.c
+++ b/server/mc-server.c
@@ -174,6 +174,8 @@ main (int argc, char **argv)
     tp_debug_set_flags (g_getenv ("MC_TP_DEBUG"));
 
     mcd = mcd_service_new ();
+    if (mcd == NULL)
+      return 1;
 
     /* Listen for suicide notification */
     g_signal_connect_after (mcd, "abort", G_CALLBACK (on_abort), mcd);
-- 
cgit v0.10.2

